File: app/src/main/java/com/example/moviesearcher/ui/details/overview/OverviewFragment.java
package com.example.moviesearcher.ui.details.overview;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import com.example.moviesearcher.R;
import com.example.moviesearcher.databinding.FragmentMovieOverviewBinding;
import com.example.moviesearcher.util.BundleUtil;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class OverviewFragment extends Fragment implements BottomNavigationView.OnNavigationItemSelectedListener {

    @BindView(R.id.information_layout) ConstraintLayout informationLayout;
    @BindView(R.id.progress_bar) ProgressBar progressBar;
    @BindView(R.id.bottom_navigation) BottomNavigationView bottomNavigationView;

    private FragmentMovieOverviewBinding fragmentView;
    private OverviewViewModel viewModel;

    public OverviewFragment(){}

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        fragmentView = DataBindingUtil.inflate(inflater, R.layout.fragment_movie_overview, container,false);
        ButterKnife.bind(this, fragmentView.getRoot());
        return fragmentView.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        bottomNavigationView.setOnNavigationItemSelectedListener(this);

        viewModel = ViewModelProviders.of(this).get(OverviewViewModel.class);
        viewModel.fetch(getActivity(), getArguments());
        observeViewModel();
    }

    private void observeViewModel(){
        viewModel.getCurrentMovie().observe(getViewLifecycleOwner(), movie -> {
            if (movie != null){
                fragmentView.setMovie(movie);
                informationLayout.setVisibility(View.VISIBLE);
            }
        });

        viewModel.getLoading().observe(getViewLifecycleOwner(), isLoading -> {
            if (isLoading != null){
                progressBar.setVisibility(isLoading ? View.VISIBLE : View.GONE);
                if(isLoading)
                    informationLayout.setVisibility(View.GONE);
            }
        });
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
        switch (menuItem.getItemId()){
            case R.id.media_menu_item:
                if (getArguments() != null) {
                    NavDirections action = OverviewFragmentDirections.actionMovieMedia(getArguments().getInt(BundleUtil.KEY_MOVIE_ID));
                    Navigation.findNavController(bottomNavigationView).navigate(action);
                }
                break;

            case R.id.cast_menu_item:
                if (getArguments() != null) {
                    NavDirections action = OverviewFragmentDirections.actionMovieCast(getArguments().getInt(BundleUtil.KEY_MOVIE_ID));
                    Navigation.findNavController(bottomNavigationView).navigate(action);
                }
                break;
        }
        return false;
    }

}

File: app/src/main/java/com/example/moviesearcher/ui/details/overview/OverviewFragment.kt
package com.example.moviesearcher.ui.details.overview

import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.example.moviesearcher.R
import com.example.moviesearcher.databinding.FragmentMovieOverviewBinding
import com.example.moviesearcher.model.data.Movie
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.fragment_movie_overview.*

class OverviewFragment : Fragment(), BottomNavigationView.OnNavigationItemSelectedListener {

    private lateinit var bottomNavigationView: BottomNavigationView

    private lateinit var fragmentView: FragmentMovieOverviewBinding
    private lateinit var viewModel: OverviewViewModel

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        fragmentView =
            DataBindingUtil.inflate(inflater, R.layout.fragment_movie_overview, container, false)
        return fragmentView.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        bottomNavigationView = bottom_navigation
        bottomNavigationView.setOnNavigationItemSelectedListener(this)
        viewModel = ViewModelProvider(this).get(OverviewViewModel::class.java)
        viewModel.fetch(arguments)
        observeViewModel()
    }

    private fun observeViewModel() {
        viewModel.currentMovie.observe(viewLifecycleOwner, Observer { movie: Movie? ->
            if (movie != null) {
                fragmentView.movie = movie
                information_layout.visibility = View.VISIBLE
            }
        })
        viewModel.loading.observe(viewLifecycleOwner, Observer { isLoading: Boolean? ->
            if (isLoading != null) {
                progress_bar.visibility =
                    if (isLoading)
                        View.VISIBLE
                    else View.GONE
                if (isLoading) information_layout.visibility = View.GONE
            }
        })
    }

    override fun onNavigationItemSelected(menuItem: MenuItem): Boolean {
        return viewModel.onNavigationItemSelected(bottomNavigationView, menuItem)
    }
}